### PR TITLE
🎉  Add a basic simulator for simple theorical tests

### PR DESCRIPTION
**Pull request by MathisFederico**

### CODE REVIEW

**Comment by MathisFederico** [comment]
Depends on #1

File: idnc/neutralization/simulation.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    step_time: float,
+    verbose: int = 0,
+):
+    """Basic simulation used for simple theorical testing.
+
+    Args:
+        n_steps (int): Number of simulation steps.
+        controller (Callable): Behavior of the robots returning their wanted velocities from all robots positions.
+        initial_poses (np.ndarray): Initial poses of the robots.
+            First dimention lenght should be the number of robots.
```
Comment (GaboFlo) [inline idnc/neutralization/simulation.py]:
```suggestion
            First dimension length should be the number of robots.
```

File: idnc/neutralization/simulation.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from typing import Tuple, Callable
+import numpy as np
+
+
+def basicsim(
+    n_steps: int,
+    controller: Callable[[np.ndarray], Tuple[np.ndarray, np.ndarray]],
+    initial_poses: np.ndarray,
+    step_time: float,
+    verbose: int = 0,
```
Comment (GaboFlo) [inline idnc/neutralization/simulation.py]: Why not a boolean  for  `verbose ` if you only use >1 condition ?

File: README.md [hunk 3]
```diff
@@ -1 +1,28 @@
-# IDNConsensus
\ No newline at end of file
+# Intruder Detection & Neutralisation Consensus (IDNC)
+
+Project to apply consensus algorithms to reproduce the defence mechanisms of the immune system.
+
+Using a consensus of cameras we detect a robot intruder (like a cell intruder) and estimate its position, then we send robots towards the estimated position (like antibodies neutralizing the threat). 
+
+This project is a colaboration between CentraleSupelec french engineering school and US students at the Rogers University.
```
Comment (GaboFlo) [inline README.md:7]:
```suggestion
This project is a collaboration between CentraleSupélec French engineering school and US students at the Rogers University.
```

**Review by GaboFlo — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (GaboFlo) [reply README.md:7]: Same as #1

File: tests/neutralization/test_simulation.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    # Expected simulation
+    expected_history = [initial_poses]
+    for step in range(n_steps):
+        last_poses = expected_history[-1]
+        expected_history.append((1 - gain * step_time) * last_poses)
+
+    # Checks
+    for step, (poses, expected) in enumerate(zip(history, expected_history)):
+        check.is_true(
+            np.all(np.abs(poses - expected) < 1e-6),
```
Comment (GaboFlo) [inline tests/neutralization/test_simulation.py:33]: Empirical threshold ?

File: idnc/neutralization/simulation.py (hunk 2 above)
Comment (MathisFederico) [reply idnc/neutralization/simulation.py]: I often end up needing multiple levels of verbose, that's why ahah
Here it's useless ... for now

File: README.md (hunk 3 above)
Comment (MathisFederico) [reply README.md:7]: Will be fixed there then

File: tests/neutralization/test_simulation.py (hunk 4 above)
Comment (MathisFederico) [reply tests/neutralization/test_simulation.py:33]: Yeah often in python "almost equal" is in this order of magnitude

**Comment by MathisFederico** [comment]
I've started to work on a rendering.
I want to add rotations handling before merging.

You can chek it out using: `python -m idnc.simulation`.

**Pull request closed without merging**